[tech] merge and document invocation (#16)
CanalTP/git_release: README.md

# generic git release script

The script performs a git release of a project

The Release process is very simple, it merges a development branch (usually called 'master' or 'dev') to the release
branch (usually called 'release'). After the merge the release branch is tagged with the new version number.

![release process](doc/release.png)

A changelog can be created using github PullRequests merged since last release.

## Invocation

Using python 3 is required (3.6 and 3.7 are working).\
It is advised to create a dedicated python virtualenv to avoid system-wise pip install.\
Cloning repository is mandatory (pip package is deprecated).

### Installation and run

The recommended way.

In the dedicated python 3 virtualenv, inside repository:

```bash
python setup.py install    # install in venv (dependencies included)
githubflow_release --help  # invoke
```

### Simple python command

Useful way for debugging.

In the dedicated python 3 virtualenv, inside repository:

```bash
pip install -r requirements.txt          # install dependencies only
python githubflow_release/run.py --help  # invoke
```

## Script usage

To know all the parameters use the --help option.

> Since for a project some options need to be consistent, you can create a `gitflow_release.yml` in the root

Reviewer: ```suggestion
Some projects may need to keep consistency for some options: you can create a `gitflow_release.yml` in the root
```
